src/zodbshootout/fork.py

```diff
@@ -32,30 +33,43 @@
     'mp': (MPProcess, MPQueue)
 }
 
-# message_delay contains the maximum expected message delay.  If a message
+# MESSAGE_DELAY contains the maximum expected message delay.  If a message
 # takes longer than this to deliver to a child process, synchronized
 # execution will fail.  OTOH, this delays test execution, so it should
 # be reasonably short.
-message_delay = 0.5
+MESSAGE_DELAY = 0.5
 
 
 class ChildProcessError(Exception):
     """A child process failed"""
 
+class UnexpectedChildDeathError(ChildProcessError):
+    """
+    A child process quit unexpectedly.
+    """
+
+class ExceptionInChildError(ChildProcessError):
+    """
+    A child process raised an exception and died.
+    """
+
 
-def run_in_child(func, strategy, *args, **kw):
-    """Call a function in a child process.  Don't return anything.
+def run_in_child(func, strategy, *args, **kwargs):
+    """
+    Call a function in a child process.
 
     Raises an exception if the child process fails.
+
+    :return: Whatever the function returned.
     """
-    Process = strategies[strategy][0]
 
-    p = Process(target=func, args=args, kwargs=kw)
-    p.start()
-    p.join()
-    if hasattr(p, 'exitcode') and p.exitcode:
-        raise ChildProcessError(
-            "process running %r failed with exit code %d" % (func, p.exitcode))
+    Process, Queue = strategies[strategy]
+
+    queue = Queue()
```

Comment:
No, that's the parent_queue (for communication to the parent). There is no child queue (communication to the child) because these functions have never been allowed to call `sync` (which is where the child queue would be used).